src/test/kotlin/io/github/gradlenexus/publishplugin/KotlinParameterizeTest.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import org.junit.jupiter.api.TestInstance
+
+/**
+ * Change the lifecycle of a test class to [PER_CLASS][TestInstance.Lifecycle.PER_CLASS]
+ * so that parameterized tests can have non-static [MethodSource][org.junit.jupiter.params.provider.MethodSource]
+ * test argument factory methods and can be used in Kotlin tests
+ *
+ * Based on: https://blog.oio.de/2018/11/13/how-to-use-junit-5-methodsource-parameterized-tests-with-kotlin/
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@Target(AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.RUNTIME)
+annotation class KotlinParameterizeTest
```

Comment:
If it works for all tests, I think we should instead configure this in `src/test/resources/junit-platform.properties` (see https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-instance-lifecycle-changing-default).

Reply:
We lose a JUnit 4 "hack" for inline field initialization per test :-)